### ISSUE TITLE
GMMHMM Unable to classify properly

### ISSUE THREAD

**Issue by leonardltk** [hmmlearn/hmmlearn#227]
**Hi, my input is as follows**

`## Training using fit()`
`n_components = 5; `
`n_mix = 16;`
`model = hmm.GMMHMM(n_components=n_components, n_mix=n_mix, tol=1e-5, n_iter=20, verbose=True) `
`## Pseudo Data with clearly seperated values`
`idx = 100`
`X1 = idx*[[1.]]`
`X2 = idx*[[100.]]`
`X3 = idx*[[200.]]`
`X4 = idx*[[300.]]`
`X5 = idx*[[400.]]`
`X = np.concatenate([X1, X2, X3, X4, X5])`
`lengths = [len(X1), len(X2), len(X3), len(X4), len(X5)]`
`model.fit(X, lengths)`
`## By right below should be split into different classes well.`
`print('predict class labels')`
`print(model.predict(X)[:100])`
`print(model.predict(X)[100:100+100])`
`print(model.predict(X)[200:200+100])`
`print(model.predict(X)[300:300+100])`
`print(model.predict(X)[400:400+100])`
`## Output Class Probabilities`
`print('\npredict_proba of each class\n', model.predict_proba(X).T)`

**However, I get the following output instead**

`predict class labels
[4 4 4 ... 4 4 4]
[2 2 2 ... 2 2 2]
[2 2 2 ... 2 2 2]
[4 4 4 ... 4 4 4]
[2 2 2 ... 2 2 2]`
`predict_proba of each class
 [[ 0.2         0.2         0.2        ...,  0.1999996   0.1999996
   0.1999996 ]
 [ 0.2         0.2         0.2        ...,  0.20000013  0.20000013
   0.20000013]
 [ 0.2         0.2         0.2        ...,  0.20000034  0.20000034
   0.20000034]
 [ 0.2         0.2         0.2        ...,  0.20000013  0.20000013
   0.20000013]
 [ 0.2         0.2         0.2        ...,  0.19999979  0.19999979
   0.19999979]]`

**I have the following output from verbose=True**

`         1       -3240.8606             +nan`
`         2       -3165.1110         +75.7496`
`         3       -3154.7477         +10.3632`
`         4       -3145.8147          +8.9331`
`         5       -3139.2282          +6.5864`
`         6       -3134.5066          +4.7216`
`         7       -3130.2161          +4.2905`
`         8       -3125.2607          +4.9554`
`         9       -3118.1095          +7.1512`
`        10       -3101.4454         +16.6641`
`        11       -3023.1916         +78.2538`
`C:\Users\ltiangku\AppData\Local\Continuum\Anaconda3\lib\site-packages\hmmlearn\hmm.py:593: RuntimeWarning: divide by zero encountered in log lgmm_posteriors = (np.log(g.predict_proba(X))`
`        12       -2382.4665        +640.7251`
`        13       -2352.4454         +30.0211`
`        14       -2257.4493         +94.9961`
`        15       -1428.2377        +829.2116`
`        16        -902.5032        +525.7344`
`        17        -902.4685          +0.0348`
`        18        -902.4648          +0.0037`
`        19        -902.4598          +0.0050`
`        20        -902.4525          +0.0072`

**Comment by vrenkens** [comment]
I took a look at the code and maybe I misuderstand, but it looks like they only use the final observation sequence (in your example X5) to update the GMM. 

If you look into _accumulate_sufficient_statistics it is just overriding the sufficient statistics instead of updating them. For example line 839 in hmmlearn/hmm.py:

`stats['samples'] = X` 

so after going over all observation sequences only the final X will be stored in stats['samples'] (correct me if I'm wrong).

For this reason I would not use the GMMHMM class.

**Comment by labodyn** [comment]
I just opened an issue noting GMMHMM doesn't work correctly when installed with pip install hmmlearn. Try installing directly from github.

**Comment by anntzer** [comment]
Should be fixed as pf ùasyer by #107, please request a reopen if not.
